### PR TITLE
add migration script and database func to move filepaths of data index

### PR DESCRIPTION
**Pull request by ifrit98**
Creates a script and python code path for miners to migrate the filepaths of their data files and redis index in the event of moving the data to a different directory than previously used for storing data.

This can be necessary if your miner went down and you recover the data to another machine with a different path, or if you simply want to move data paths and the index does not reflect this change.

If you need to migrate data AND index, run:
```bash
bash scripts/migrate_database_directory.sh <OLD_PATH> <NEW_PATH> <DATABASE_INDEX>
```

Example usage:

```bash
bash scripts/migrate_database_directory.sh ~/.data ~/.new_data_path 0 
```

If you need to ONLY update the index filepaths, run:
```
scripts/migrate_miner_data.py --help

usage: migrate_miner_data.py [-h] [--database_index DATABASE_INDEX] --new_data_directory NEW_DATA_DIRECTORY

options:
  -h, --help            show this help message and exit
  --database_index DATABASE_INDEX
  --new_data_directory NEW_DATA_DIRECTORY
``` 

This will be accompanied by a follow-up PR to fix this on the miner side directly and automatically account for this in the future.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -515,6 +515,32 @@ pm2 start /home/user/storage-subnet/neurons/miner.py --interpreter /home/user/mi
 These options allow you to configure the miner's behavior, database connections, blacklist/whitelist settings, priority handling, and integration with monitoring tools like WandB. Adjust these settings based on your mining setup and requirements.
 
 
+#### Data migration
```
Comment (eduardogr) [inline README.md:518]: Do we need to add this to the README's index?

File: README.md [hunk 2]
```diff
@@ -515,6 +515,32 @@ pm2 start /home/user/storage-subnet/neurons/miner.py --interpreter /home/user/mi
...
+If for whatever reason you need to migrate the data in your hard drive configured with `--database.directory` to a new directory than is reflected in the Redis index, then you can do it by simply running the script in `scripts/migrate_miner_data.py`.
```
Comment (eduardogr) [inline README.md:519]: I think we could enumerate when is the migration needed. At least the scenarios we have identified so far,

what do you think about that?

Comment (eduardogr) [reply README.md:519]: e.g:

```
When should you run the migration script?:
- if you want to specify a different --database.directory
```

File: README.md (hunk 1 above)
Comment (ifrit98) [reply README.md:518]: I think it's ok to leave. Don't want to create a 3rd level of nesting, as 4 #s (####) are not captured elsewhere.

File: README.md (hunk 2 above)
Comment (ifrit98) [reply README.md]: Yes, good idea. Adding this now.

**Pull request merged**